Rdfg enhancements

rawstorage opened the pull request:
Replication Enhancements
-------------------------
RDF functions added
-------------------
-create_rdf_group
-modify_rdf_group
-delete_rdf_group
-get_rdf_port_connections
-get_rdf_director_port_details
-get_rdf_director_ports
-get_rdf_director
-get_rdf_director_list
-create_storage_group_from_rdfg

BugFix
---------
update to rest_requests header parameter corrected to application-type line 54

Fix in CI_Test for effective_wwn functions and set initiator flags

rawstorage committed:
added new functions for replication.py and bug file for application-type header incorrectly in lowercase.  Added skip to 2 tests in provisioning ci_tests for external wwn.

rawstorage committed:
added new functions for replication.py and bug file for application-type header incorrectly in lowercase.  Added skip to 2 tests in provisioning ci_tests for external wwn.

rawstorage committed:
maded port detail port_details to be consistent with other module naming.

rawstorage committed:
pep8 error line1123 replication, spotted after commit. Over indendation thanks to extra space with pluralisation. :(

rawstorage committed:
added new function create_storage_group_from_rdfg with associated unit test and ci_test + update to change log.

odonos12 committed:
Added same delimiter and quotechar params to
csv reader that are used in csv writer. Added skips
to effective wwn tests where no effective wwn enabled
volumes can be found.

odonos12 committed:
Added same delimiter and quotechar params to
csv reader that are used in csv writer. Added skips
to effective wwn tests where no effective wwn enabled
volumes can be found.

odonos12 committed:
Added additional setup to test_modify_initiator_change_initiator_flags
test to prevent issues where the in-use initiator that was selected
was in a group causing conflicts when attempting to set flags.

odonos12 committed:
Added additional setup to test_modify_initiator_change_initiator_flags
test to prevent issues where the in-use initiator that was selected
was in a group causing conflicts when attempting to set flags.

rawstorage committed:
updated unit_test for application-type header.

added create_storage_group_from_rdfg ci_test and unit test

rawstorage committed:
Merge remote-tracking branch 'origin/rdfg_enhancements' into rdfg_enhancements

helenwalsh commented on PyU4V/replication.py:
```diff
@@ -11,7 +11,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""replication.py."""
+'''replication.py.'''
```
Doc strings should be in double quotes

helenwalsh commented on PyU4V/replication.py:
```diff
@@ -46,13 +46,17 @@
 
 class ReplicationFunctions(object):
-    """ReplicationFunctions."""
+    '''ReplicationFunctions.'''
```
Looks like they have been changed throughout the document.  Please change them to ".  Thanks

rawstorage replied:
Thanks Helen that was a find replace error.

helenwalsh commented on PyU4V/replication.py:
```diff
@@ -863,38 +867,38 @@ def modify_storage_group_srdf(
-        Optional boolean parameters to set are "bypass", "metroBias", "star",
-        "immediate", "hop2", "consExempt", "force", "symForce".
+        Optional boolean parameters to set are 'bypass', 'metroBias', 'star',
+        'immediate', 'hop2', 'consExempt', 'force', 'symForce'.
```
Just to avoid confusion.  Having single quotes here is fine. It is also in the doc strings so less concerned with this anyway

rawstorage replied:
I've changed back to original, this change was made in error

helenwalsh commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,241 @@ def delete_storage_group_srdf(
+
+        :param array_id: 12 digit serial number for PowerMax array -- str
+        :param director_id: identifier for director e.g. RF-1F -- str
+        :param filters: optional filters - dict
```
nit -- instead of -

rawstorage replied:
done

helenwalsh commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,241 @@ def delete_storage_group_srdf(
+        :param array_id: 12 digit serial number for  PowerMax array -- str
+        :param director_id: identifier for director e.g. RF-1F -- str
+        :param port_id: port number -- int
+        :returns: port details --dict
```
nit - add a stace

rawstorage replied:
good eyes

helenwalsh commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,241 @@ def delete_storage_group_srdf(
+        :param array_id: 12 digit serial number for Source  -- str
+        :param director_id: identifier for director e.g. RF-1F -- str
+        :param port_id: port number -- int
+        :returns: remote port details --dict
```
nit, add a space

MichaelMcAleer replied:
Still needs a space

helenwalsh commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,241 @@ def delete_storage_group_srdf(
+        :param dev_list: list of volumes to be moved between RDF groups -- list
+        :param target_srdf_group: rdfg group to move volumes to -- int
+        :param consistency_exempt: ignore device for consistency checks--bool
+        :returns:
```
If it doesn't return anything, remove :returns:

helenwalsh commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,241 @@ def delete_storage_group_srdf(
+        :param consistency_exempt: ignore device for consistency checks--bool
+        :returns:
+        '''
+        rdfg_actions = {'MOVE': 'Move', 'ADD_PORTS': 'add_ports',
```
Should 'MOVE', 'ADD_PORT' etc., the keys of the dict be constants?

MichaelMcAleer replied:
What Helen said, change these to constants

helenwalsh commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,241 @@ def delete_storage_group_srdf(
+
+        :param srdf_group_number: number of RDF group to be deleted -- int
+        :param array_id: 12 digit serial of array -- str
+        :returns:
```
remove :returns: if nothing is returned from the function

helenwalsh commented on PyU4V/tests/ci_tests/base.py:
```diff
@@ -191,6 +193,79 @@ def cleanup_rdfg(self, sg_name, srdf_group_number):
+        local_online_rdf_ports = []
+        remote_online_rdf_ports = []
+        for director in local_rdf_director_list:
+            local_rdf_director_port_list = \
```
brackets instead of \

MichaelMcAleer replied:
Still needs addressed

helenwalsh commented on PyU4V/tests/ci_tests/base.py:
```diff
@@ -191,6 +193,79 @@ def cleanup_rdfg(self, sg_name, srdf_group_number):
+        """Gets a list of all RDF ports online for local and remote array.
+
+        :returns: list of online ports rdf ports
+
```
remove line

helenwalsh commented on PyU4V/tests/ci_tests/base.py:
```diff
@@ -191,6 +193,79 @@ def cleanup_rdfg(self, sg_name, srdf_group_number):
+        remote_rdf_director_list = self.replication.get_rdf_director_list(
+            filters={'online': True})
+        for director in remote_rdf_director_list:
+            remote_rdf_director_port_list = \
```
brackets instead of \

rawstorage replied:
done

helenwalsh commented on PyU4V/tests/ci_tests/base.py:
```diff
@@ -191,6 +193,79 @@ def cleanup_rdfg(self, sg_name, srdf_group_number):
+        """Helper function to get RDFG on arrays that is free for use
+
+        :returns: next RDF group number on local and remote array
+
```
remove empty line

rawstorage replied:
done, found another like this

helenwalsh commented on PyU4V/tests/ci_tests/test_pyu4v_ci_replication.py:
```diff
@@ -482,3 +483,111 @@ def test_failback_storage_group_srdf(self):
             storage_group_id=sg_name, rdfg_num=srdf_group_number).get('states')
         self.assertIn('Synchronized', status)
+
+    def test_get_rdf_director_list(self):
```
nit, add a description.  It doesn't look like the version of our pep8 complains anymore but just to be consistent with the rest of the test file

helenwalsh commented on PyU4V/tests/ci_tests/test_pyu4v_ci_replication.py:
```diff
@@ -482,3 +483,111 @@ def test_failback_storage_group_srdf(self):
+        self.assertIsInstance(rdf_director_list, list)
+
+    def test_get_rdf_director_detail(self):
+        rdf_director_list = self.replication.get_rdf_director_list()
```
same

helenwalsh commented on PyU4V/tests/unit_tests/pyu4v_common_data.py:
```diff
@@ -11,11 +11,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""pyu4v_common_data.py."""
+'''pyu4v_common_data.py.'''
```
double quotes

rawstorage replied:
done

helenwalsh reviewed (changes requested):
A few style issues only.  Also, just for future reference,  perhaps we separate the issues into pull requests so for example application-type fix and initiator group tests fix get their own pull request.  Just a suggestion.

rawstorage committed:
corrected quotes in docstrings

rawstorage committed:
corrected quotes in docstrings

rawstorage committed:
Changes per review on Pull 84

rawstorage committed:
Changes per review on Pull 84

rawstorage committed:
Changes per review on Pull 84

rawstorage committed:
Changes per review on Pull 84

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+
+        :param array_id: 12 digit serial number for PowerMax array -- str
+        :param director_id: identifier for director e.g. RF-1F -- str
+        :returns: director details --dict
```
params are in the incorrect order, returns should have a space between --  and dict

rawstorage replied:
done

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+            resource_level=SYMMETRIX, resource_level_id=array_id,
+            resource_type=RDF_DIRECTOR, resource_type_id=director_id,
+            resource=PORT, params=filters
+        )
```
Nit: move the ) to the previous line instead of a line of its own

rawstorage replied:
done

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+            resource_level=SYMMETRIX, resource_level_id=array_id,
+            resource_type=RDF_DIRECTOR, resource_type_id=director_id,
+            resource=PORT, resource_id=port_id
+        )
```
Nit: Move ) to previous line

rawstorage replied:
done

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+        :param array_id: 12 digit serial number for PowerMax array -- str
+        :param director_id: identifier for director e.g. RF-1F -- str
+        :param filters: optional filters -- dict
+        :returns: list of RDF ports -- list
```
Params are in incorrect order

rawstorage replied:
resorted

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+
+        :param array_id: 12 digit serial number for  PowerMax array -- str
+        :param director_id: identifier for director e.g. RF-1F -- str
+        :param port_id: port number -- int
```
Params are in incorrect order

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+
+        :param array_id: 12 digit serial number for Source  -- str
+        :param director_id: identifier for director e.g. RF-1F -- str
+        :param port_id: port number -- int
```
Params are in incorrect order

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+        :param remote_array_id: 12 digit serial number of remote array  -- str
+        :param remote_rdfg_number rdfg for the remote array -- int
+        :param label: Label for group up to 10 characters -- str
+        :param remote_director_port_list: list of remote directors and ports to
```
Params are in incorrect order and the formatting seems to be off, multi lines should be in line with the start of text in  sentence above

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+        :param label: Label for group up to 10 characters -- str
+        :param remote_director_port_list: list of remote directors and ports to
+                                          group e.g [RF-1E:1, RF-2E:1] -- list
+        :returns:
```
remove returns, there is no return

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+                         [RF-1E:10, RF-2E:10] --list
+        :param dev_list: list of volumes to be moved between RDF groups -- list
+        :param target_srdf_group: rdfg group to move volumes to -- int
+        :param consistency_exempt: ignore device for consistency checks--bool
```
Params are in incorrect order

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+                    'targetRdfGroup': target_rdf_group,
+                    'volumesToMove': dev_list,
+                    'exempt': consistency_exempt
+                },
```
move } to previous line

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+                    "label": label
+                },
+                "action": "set_label"
+            }
```
Move both } to previous line, this reduces the overall length of the file and makes our IDEs load inspections faster

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+                "action": "set_label"
+            }
+
+        elif rdfg_action == 'add_ports' or 'remove_ports':
```
This wont work, second half of or statement will always evaluate to True because it is a populated string. Should be:

`elif rdfg_action == 'add_ports' or rdfg_action == 'remove_ports':`

A better implementation would be

`elif rdfg_action in ['add_ports', 'remove_ports']:`

rawstorage replied:
changed will need to re-run ci to verify still working

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+                             resource_level_id=array_id,
+                             resource_type=RDF_GROUP,
+                             resource_type_id=srdf_group_number,
+                             payload=payload)
```
Collapse these into a few lines, save space where possible

rawstorage replied:
done

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+                             resource_level_id=array_id,
+                             resource_type=RDF_GROUP,
+                             resource_type_id=srdf_group_number,
+                             )
```
Move ) to previous line, collapse params to as few lines as possible

rawstorage replied:
done and collapsed

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+    def create_storage_group_from_rdfg(self, storage_group_name,
+                                       srdf_group_number, array_id=None,
+                                       rdf_type=None,
+                                       remote_storage_group_name=None):
```
Collapse to fewer lines

rawstorage replied:
done

MichaelMcAleer replied:
Input params still over 4 lines

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,281 @@ def delete_storage_group_srdf(
+                         added to the Storage Group. Only needs to be populated
+                         if the SRDF group contains both RDF1 and RDF2 volumes.
+                         valid values RDF1 or RDF2-- str
+        :param remote_storage_group_name: name of remote storage group--str
```
Params dont match up

rawstorage replied:
done, re-ran pep8 all clean

MichaelMcAleer reviewed (changes requested):
I stopped reviewing this, too much to add in one review.

Fix the parameters, their ordering, and collapse all the multi-line stuff to as few lines as possible to save on space in the overall file, it helps out IDEs load inspections faster. There is also a lot of trailing ) or } which could be moved up a line or two to further save space.

After these are done ill go through the code for problems

rawstorage committed:
Changes per review on Pull 84

rawstorage committed:
Changes per review on Pull 84

rawstorage committed:
Ordering of Parameters in new function doc strings Changes per review on Pull 84

rawstorage committed:
Ordering of Parameters in new function doc strings Changes per review on Pull 84

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+        or more remote ports for the desired target array and feed into this
+        function. Additional Ports can be added with modify_rdf_group function.
+
+        :param local_director_Port_list: list of local directors and ports for
```
lower case chars only

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+                                          group e.g [RF-1E:1, RF-2E:1] -- list
+        :param array_id: array_id: 12 digit serial number of Source (R1) array,
+                        if no array is specified the array in config file or
+                        api connection will be default -- str
```
Second and third lines need to move over a space to align with first line

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+            dir_id = port.split(':')[0]
+            port_no = port.split(':')[1]
+            remote_ports.append({'symmetrixID': remote_array_id,
+                                 'directorId': dir_id, 'portNumber': port_no})
```
Nit: It would be good to keep symmetrixID and similar keys to constants, saves us in the future so we only need to change constants in one place if they change

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+
+        :param action: add_ports, remove_ports, move
+        :param srdf_group_number: srdf group number
+                                  for action on local array: int
```
not sure if a second line is needed here, and -- should be used instead of : when defining input type

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+                                  for action on local array: int
+        :param array_id: 12 digit serial of array -- str
+        :param port_list: list of ports to be added or removed e.g.
+                         [RF-1E:10, RF-2E:10] --list
```
second line needs to be moved over one space to align with first line beginning

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+        :param array_id: 12 digit serial of array -- str
+        :param port_list: list of ports to be added or removed e.g.
+                         [RF-1E:10, RF-2E:10] --list
+        :param label: Label for group up to 10 characters -- str
```
Nit: description starts upper case char whilst all others start lower case char

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+        :param label: Label for group up to 10 characters -- str
+        :param dev_list: list of volumes to be moved between RDF groups -- list
+        :param target_srdf_group: rdfg group to move volumes to -- int
+        :param consistency_exempt: ignore device for consistency checks--bool
```
spaces required between checks -- bool

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+
+        elif rdfg_action in ['add_ports', 'remove_ports']:
+            if not port_list:
+                msg = ('list of ports must be supplied when adding or removing'
```
there is no space at the end of the first line so when printed to logs or screen it will be 'removingports' instead of 'removing ports'

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+
+        create_sg_payload = {
+            'rdf_group_number': srdf_group_number,
+        }
```
comma after last variable, move } back to previous line

MichaelMcAleer commented on PyU4V/replication.py:
```diff
@@ -1062,3 +1066,267 @@ def delete_storage_group_srdf(
+            'create_sg_from_rdfg': create_sg_payload,
+            'storage_group_name': storage_group_name,
+            'action': 'CreateSgFromRdfg'
+        }
```
move } back to previous line

MichaelMcAleer commented on PyU4V/tests/ci_tests/test_pyu4v_ci_provisioning.py:
```diff
@@ -568,7 +569,22 @@ def test_modify_initiator_set_fcid(self):
+            port_group_name, masking_view_name, storage_group_name, host)
+        self.addCleanup(self.delete_masking_view, masking_view_name)
+        initiator_list = self.provisioning.get_initiator_list()
+        initiator = [x for x in initiator_list if initiator_name in x][0]
```
This is potentially dangerous? If initiator_name is not matched you will try to get the first element in an empty list

rawstorage replied:
@odonos12 can you re-examine this change

odonos12 replied:
At this point we've already created a host & masking view with an initiator (initiator_name) set, is there still a possibility that get_initiator_list() would still return a list that doesn't contain our initiator?

MichaelMcAleer replied:
You could assert the list is populated then extract the initiator name afterwards, but if you are confident that we wont ever run into that scenario where the initiator name isn't in the list then it will be fine as is

rawstorage replied:
@odonos12  are you ok to mark this as resolved?

MichaelMcAleer commented on PyU4V/tests/unit_tests/pyu4v_common_data.py:
```diff
@@ -239,15 +277,15 @@ class CommonData(object):
-                       'snapvx_target': 'false',
-                       'snapvx_source': 'false',
+                       'snapvx_target': 'False',
+                       'snapvx_source': 'False',
```
These should not have changed, our REST API expects lower case string 'false' and 'true', same as settings above, if you are breaking tests then you need to resolve the function and not change the common data to suit the new functions

MichaelMcAleer commented on PyU4V/tests/unit_tests/test_pyu4v_replication.py:
```diff
@@ -486,3 +486,86 @@ def test_delete_storagegroup_srdf_no_number(self):
                 resource_type_id=self.data.storagegroup_name,
                 resource='rdf_group', resource_id=self.data.rdf_group_no)
+
+    def test_create_rdf_group(self):
```
Description needed

MichaelMcAleer commented on PyU4V/tests/unit_tests/test_pyu4v_replication.py:
```diff
@@ -486,3 +486,86 @@ def test_delete_storagegroup_srdf_no_number(self):
+                label='test')
+            mock_create.assert_called_once()
+
+    def test_modify_rdf_group_remove_port(self):
```
Description needed

MichaelMcAleer commented on PyU4V/tests/unit_tests/test_pyu4v_replication.py:
```diff
@@ -486,3 +486,86 @@ def test_delete_storagegroup_srdf_no_number(self):
+                port_list=['RF-1E:1'])
+            mock_mod.assert_called_once()
+
+    def test_modify_rdf_group_add_port(self):
```
Description needed

MichaelMcAleer commented on PyU4V/tests/unit_tests/test_pyu4v_replication.py:
```diff
@@ -486,3 +486,86 @@ def test_delete_storagegroup_srdf_no_number(self):
+                port_list=['RF-1E:1'])
+            mock_mod.assert_called_once()
+
+    def test_modify_rdf_group_set_label(self):
```
Description needed

MichaelMcAleer commented on PyU4V/tests/unit_tests/test_pyu4v_replication.py:
```diff
@@ -486,3 +486,86 @@ def test_delete_storagegroup_srdf_no_number(self):
+                label='new_label')
+            mock_mod.assert_called_once()
+
+    def test_delete_rdf_group(self):
```
Description needed

MichaelMcAleer commented on PyU4V/tests/unit_tests/test_pyu4v_replication.py:
```diff
@@ -486,3 +486,86 @@ def test_delete_storagegroup_srdf_no_number(self):
+
+    @mock.patch.object(common.CommonFunctions, 'get_request',
+                       return_value=pcd.CommonData.remote_port_details)
+    def test_get_rdf_port_remote_connections(self, mck_get):
```
Description needed

MichaelMcAleer commented on PyU4V/tests/unit_tests/test_pyu4v_replication.py:
```diff
@@ -486,3 +486,86 @@ def test_delete_storagegroup_srdf_no_number(self):
+
+    @mock.patch.object(common.CommonFunctions, 'get_request',
+                       return_value=pcd.CommonData.rdf_dir_port_detail)
+    def test_get_rdf_director_port_details(self, mck_get):
```
Description needed, and all other functions below in this file

MichaelMcAleer commented on PyU4V/utils/file_handler.py:
```diff
@@ -64,7 +64,7 @@ def _convert(s):
     with open(file_name, newline='') as infile:
         # read the file as a dictionary for each row ({header : value})
-        reader = csv.DictReader(infile)
+        reader = csv.DictReader(infile, delimiter=',', quotechar='|')
```
If we are going to set the delimiter and quote char these should be user configurable in the function input params

rawstorage replied:
@odonos12 can you revisit this one.

MichaelMcAleer reviewed (changes requested):
More changes required there, nothing major its almost all formatting

rawstorage committed:
Addressed comments from @Michael McAleer. 2 Outstanding for Simon to take a look at

rawstorage committed:
Merge remote-tracking branch 'origin/rdfg_enhancements' into rdfg_enhancements

# Conflicts:
#	PyU4V/replication.py

MichaelMcAleer reviewed (commented):
Everything looks good Paul, once the CSVReader delimiter comment is addressed ill give it my +1

odonos12 committed:
Change delimiter and quotechar in csv reader and writer to
be assigned by function parameters.

MichaelMcAleer reviewed (approved):
Approving, all requested changes made and CI has had 4 clean runs since most recent commit.

rawstorage committed:
added docs for version 9.2, minor changes to docstrings to correct formatting for documentation

rawstorage committed:
Merge remote-tracking branch 'origin/rdfg_enhancements' into rdfg_enhancements

rawstorage committed:
added docs for version 9.2, minor changes to docstrings to correct formatting for documentation, one minor typo and additional content added to create_storage_group_from_rdf_group

rawstorage committed:
added docs for version 9.2, minor changes to docstrings to correct formatting for documentation, one minor typo and additional content added to create_storage_group_from_rdf_group,

configuration.rst -> line 152: arguements should be arguments
reccomendations.rst -> line 13: it's should be its
support.rst -> line 4: Github should be GitHub
tools.rst -> lines 37, 41, 73: Github should be GitHub
